### PR TITLE
[mlir][test] Add integration tests for vector.interleave

### PR DESCRIPTION
**Pull request by MacDue**

### CODE REVIEW

File: mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-scalable-interleave.mlir [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// RUN: mlir-opt %s -test-lower-to-llvm | \
+// RUN: %mcr_aarch64_cmd -e entry -entry-point-result=void  \
+// RUN:   -shared-libs=%mlir_c_runner_utils,%mlir_arm_runner_utils | \
+// RUN: FileCheck %s
+
+func.func @entry() {
+  %f1 = arith.constant 1.0 : f32
+  %f2 = arith.constant 2.0 : f32
+  %v1 = vector.splat %f1 : vector<[4]xf32>
+  %v2 = vector.splat %f2 :  vector<[4]xf32>
```
Comment (c-rhodes) [inline mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-scalable-interleave.mlir:10]:
```suggestion
  %v2 = vector.splat %f2 : vector<[4]xf32>
```

File: mlir/test/Integration/Dialect/Vector/CPU/test-interleave.mlir [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// RUN: mlir-opt %s -test-lower-to-llvm | \
+// RUN: mlir-cpu-runner -e entry -entry-point-result=void  \
+// RUN:   -shared-libs=%mlir_c_runner_utils | \
+// RUN: FileCheck %s
+
+func.func @entry() {
+  %f1 = arith.constant 1.0 : f32
+  %f2 = arith.constant 2.0 : f32
+  %v1 = vector.splat %f1 : vector<2x4xf32>
+  %v2 = vector.splat %f2 :  vector<2x4xf32>
```
Comment (c-rhodes) [inline mlir/test/Integration/Dialect/Vector/CPU/test-interleave.mlir:10]:
```suggestion
  %v2 = vector.splat %f2 : vector<2x4xf32>
```

**Review by c-rhodes — APPROVED** [review]
couple of whitespaces that can be removed but otherwise LGTM cheers

**Review by dcaballe — APPROVED** [review]

**Comment by MacDue** [comment]
https://lab.llvm.org/buildbot/#/builders/179/builds/9438

I missed `-march=aarch64 -mattr=+sve` of the SVE test, I'll push a fix shortly.

**Pull request merged**